daemon/create.go

```diff
@@ -324,24 +322,29 @@ func verifyNetworkingConfig(nwConfig *networktypes.NetworkingConfig) error {
 		return fmt.Errorf("container cannot be connected to network endpoints: %s", strings.Join(l, ", "))
 	}
 
+	var errs []error
 	for k, v := range nwConfig.EndpointsConfig {
 		if v == nil {
-			return fmt.Errorf("no EndpointSettings for %s", k)
+			errs = append(errs, fmt.Errorf("no EndpointSettings for %s", k))
+			continue
 		}
-		if v.IPAMConfig != nil {
-			if v.IPAMConfig.IPv4Address != "" && net.ParseIP(v.IPAMConfig.IPv4Address).To4() == nil {
-				return fmt.Errorf("invalid IPv4 address: %s", v.IPAMConfig.IPv4Address)
-			}
-			if v.IPAMConfig.IPv6Address != "" {
-				n := net.ParseIP(v.IPAMConfig.IPv6Address)
-				// if the address is an invalid network address (ParseIP == nil) or if it is
-				// an IPv4 address (To4() != nil), then it is an invalid IPv6 address
-				if n == nil || n.To4() != nil {
-					return fmt.Errorf("invalid IPv6 address: %s", v.IPAMConfig.IPv6Address)
-				}
-			}
+
+		// The referenced network k might not exist when the container is created, so don't fail if that's the case.
+		// If any other error happen, it's not a validation error, so better early exit.
+		nw, err := daemon.FindNetwork(k)
+		if err != nil && !errdefs.IsNotFound(err) {
+			return err
```

Comment:
What error can we expect here, and would we still have a network?

oh.. I guess that's the case you described 🤦 I that's not related to lazy creating networks, I'm on the fence of being ok erroring out for that